src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,129 @@
+package duke;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+
+/**
+ * Storage class handles the reading and writing of data
+ * from the local file system.
+ */
+public class Storage {
+    private DateTimeFormatter dateTimeFormat;
+    private File file;
+
+    /**
+     * Public constructor to initialise the storage for the Duii Bot.
+     */
+    public Storage() {
+        this.dateTimeFormat = DateTimeFormatter.ofPattern("yyyy MMM dd HH:mm");
+    }
+
+    /**
+     * Searches the current file system for existing saved files.
+     * Creates a new data directory consisting of the text file to write
+     * into if saved file does not exist.
+     *
+     * @return The abstract representation of the pathname of the created file
+     */
+    public File getfiles() {
```

Comment:
```suggestion
    public File getFiles() {
```

Would it be good to change the method name to follow camelCase convention? This is outlined in the module's [Naming](https://se-education.org/guides/conventions/java/intermediate.html#naming) conventions.